src/v/cloud_storage/inventory/ntp_hashes.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "model/fundamental.h"
+#include "utils/retry_chain_node.h"
+
+#include <seastar/core/iostream.hh>
+
+#include <absl/container/node_hash_set.h>
+
+namespace cloud_storage::inventory {
+
+enum class lookup_result {
+    exists,
+    missing,
+    possible_collision,
+};
+
+class ntp_path_hashes {
+public:
+    ntp_path_hashes(model::ntp ntp, std::filesystem::path hash_store_path);
+
+    ss::future<bool> load_hashes();
+
+    lookup_result exists(const remote_segment_path& path) const;
+
+private:
+    ss::future<> load_hashes(ss::input_stream<char>& stream);
+
+    model::ntp _ntp;
+    std::filesystem::path _hash_store_path;
```

Comment:
These seem unused for now? Do we need them if we're already storing _ntp_hashes_path?

Reply:
removed this field